src/framework/app.ts

```diff
@@ -297,15 +300,26 @@ export function createApp(appConfig?: { types?: any }): App {
           schema,
         })
 
-        const app = express()
+        const expressApp = express()
+        apolloServer.applyMiddleware({ app: expressApp })
 
-        server.applyMiddleware({ app })
+        return new Promise(resolve => {
+          const server = expressApp.listen(mergedConfig.port, () => {
+            const startMessage = mergedConfig.startMessage(mergedConfig.port)
 
-        app.listen({ port: mergedConfig.port }, () =>
-          console.log(mergedConfig.startMessage(mergedConfig.port))
-        )
+            if (startMessage.length > 0) {
+              console.log(startMessage)
+            }
 
-        sendServerReadySignalToDevModeMaster()
+            sendServerReadySignalToDevModeMaster()
+            httpServer = server
+            return resolve()
+          })
+        })
+      },
+      async stop() {
+        await apolloServer?.stop()
```

Comment:
What does `stop` do if not closing the server?

Reply:
https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-core/src/ApolloServer.ts#L575

subscriptionServer + Apollo engine related stuff

Reply:
Might be worth a comment, useful info!